dotCMS/src/integration-test/java/com/dotcms/publishing/job/SiteSearchJobImplTest.java

```diff
@@ -548,17 +713,118 @@ public void test_MultilangContent_IndexingAllLanguages()
+                Assert.assertEquals("Content in default Language gives NO results", 0,
+                        searchResults.getTotalResults());
+            }
+
+            if(testCase.expectedResultsWhenSearchingContentInSecondLanguage) {
+                SiteSearchResults searchResults = siteSearchAPI.search(newIndexName, "catalina", 0, 10);
+                Assert.assertTrue("Content in second Language gives results",
+                        searchResults.getTotalResults() >= 1);
+                Assert.assertEquals(pageDefaultLang.getTitle(),
+                        searchResults.getResults().get(0).getTitle());
+                Assert.assertEquals(pageDefaultLang.getLanguageId(),
+                        searchResults.getResults().get(0).getLanguage());
+            } else {
+                SiteSearchResults searchResults = siteSearchAPI.search(newIndexName, "catalina", 0, 10);
+                Assert.assertEquals("Content in second Language gives NO results", 0,
+                        searchResults.getTotalResults());
+            }
+
+            if(testCase.expectedResultsWhenSearchingContentInThirdLanguage) {
+                SiteSearchResults searchResults = siteSearchAPI.search(newIndexName, "caterina", 0, 10);
+                Assert.assertTrue("Content in third Language gives results",
+                        searchResults.getTotalResults() >= 1);
+                Assert.assertEquals(pageDefaultLang.getTitle(),
+                        searchResults.getResults().get(0).getTitle());
+                Assert.assertEquals(pageDefaultLang.getLanguageId(),
+                        searchResults.getResults().get(0).getLanguage());
+            } else {
+                SiteSearchResults searchResults = siteSearchAPI.search(newIndexName, "caterina", 0, 10);
+                Assert.assertEquals("Content in third Language gives NO results", 0,
+                        searchResults.getTotalResults());
+            }
 
         } finally {
             Config.setProperty("DEFAULT_CONTENT_TO_DEFAULT_LANGUAGE",
                     defaultContentToDefaultLangOriginalValue);
+
+            Config.setProperty("DEFAULT_PAGE_TO_DEFAULT_LANGUAGE",
+                    defaultPagetoDefaultLangOriginalValue);
         }
     }
 
+    private HTMLPageAsset createNewVersionHtmlPage(HTMLPageAsset pageDefaultLang, Language secondLang)
+            throws DotSecurityException, DotDataException {
+        Contentlet pageSecondLang = contentletAPI.find(pageDefaultLang.getInode(), systemUser, false);
+        pageSecondLang.setInode("");
+        pageSecondLang.setLanguageId(secondLang.getId());
+        pageSecondLang = contentletAPI.checkin(pageSecondLang, systemUser, false);
+        ContentletDataGen.publish(pageSecondLang);
+        return APILocator.getHTMLPageAssetAPI().fromContentlet(pageSecondLang);
+    }
+
+    private void createNewVersionAndPublishExistingEmployeeContent(Language newLang,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'newLang' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5303548)